### PR TITLE
Change CecilSymbolHelper to instance class

### LINKED ISSUE #774: Change `CecilSymbolHelper` as instance class and inject with DI
We don't like static stuff, we should avoid it, are not testable and more subject to race condition issues.
We should move `CecilSymbolHelper` to instance class extract an interface with only `GetBranchPoints` and `SkipNotCoverableInstruction` methods  and inject like other services.
We should remove static memebers https://github.com/tonerdo/coverlet/blob/master/src/coverlet.core/Symbols/CecilSymbolHelper.cs#L22

https://github.com/tonerdo/coverlet/blob/master/src/coverlet.core/Symbols/CecilSymbolHelper.cs#L19

### PR DESCRIPTION
**Pull request by daveMueller**
closes #774

### CODE REVIEW

File: src/coverlet.collector/DataCollection/CoverageManager.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using Coverlet.Collector.Utilities;
 using Coverlet.Collector.Utilities.Interfaces;
 using Coverlet.Core;
+using coverlet.core.Abstractions;
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/DataCollection/CoverageManager.cs:9]: Fix namespace casing please

File: src/coverlet.core/Abstractions/ICecilSymbolHelper.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using Coverlet.Core.Symbols;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace coverlet.core.Abstractions
+{
+    internal interface ICecilSymbolHelper
+    {
+        List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinition);
+
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Abstractions/ICecilSymbolHelper.cs:13]: nit: extra line

File: src/coverlet.core/Abstractions/ICecilSymbolHelper.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+using Coverlet.Core.Symbols;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace coverlet.core.Abstractions
+{
+    internal interface ICecilSymbolHelper
+    {
+        List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinition);
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Abstractions/ICecilSymbolHelper.cs:12]: Better return `IReadOnlyList<T>`

File: src/coverlet.core/Instrumentation/Instrumenter.cs [hunk 4]
```diff
@@ -5,7 +5,7 @@
 using System.IO;
 using System.Linq;
 using System.Runtime.CompilerServices;
-
+using coverlet.core.Abstractions;
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Instrumentation/Instrumenter.cs:8]: Add empty line after `System...`

File: src/coverlet.core/Symbols/CecilSymbolHelper.cs [hunk 5]
```diff
@@ -16,12 +16,12 @@
 
 namespace Coverlet.Core.Symbols
 {
-    internal static class CecilSymbolHelper
+    internal class CecilSymbolHelper : ICecilSymbolHelper
     {
         private const int StepOverLineCode = 0xFEEFEE;
         private static ConcurrentDictionary<string, int[]> CompilerGeneratedBranchesToExclude = null;
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Symbols/CecilSymbolHelper.cs:22]: Remove `static` semantic is the same with singleton.

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 6]
```diff
@@ -4,10 +4,12 @@
 using System.Linq;
 using System.Reflection;
 using System.Runtime.InteropServices;
-
+using coverlet.core.Abstractions;
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs:7]: Add empty line after `using System...`

File: src/coverlet.collector/DataCollection/CoverageManager.cs (hunk 1 above)
Comment (daveMueller) [reply src/coverlet.collector/DataCollection/CoverageManager.cs]: Done

File: src/coverlet.core/Abstractions/ICecilSymbolHelper.cs (hunk 2 above)
Comment (daveMueller) [reply src/coverlet.core/Abstractions/ICecilSymbolHelper.cs]: Done

File: src/coverlet.core/Abstractions/ICecilSymbolHelper.cs (hunk 3 above)
Comment (daveMueller) [reply src/coverlet.core/Abstractions/ICecilSymbolHelper.cs]: Done

File: src/coverlet.core/Instrumentation/Instrumenter.cs (hunk 4 above)
Comment (daveMueller) [reply src/coverlet.core/Instrumentation/Instrumenter.cs]: Done

File: src/coverlet.core/Symbols/CecilSymbolHelper.cs (hunk 5 above)
Comment (daveMueller) [reply src/coverlet.core/Symbols/CecilSymbolHelper.cs]: Done

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs (hunk 6 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done

File: src/coverlet.core/Symbols/CecilSymbolHelper.cs [hunk 7]
```diff
@@ -16,15 +16,15 @@
...
         private const int StepOverLineCode = 0xFEEFEE;
-        private static ConcurrentDictionary<string, int[]> CompilerGeneratedBranchesToExclude = null;
+        private readonly ConcurrentDictionary<string, int[]> _compilerGeneratedBranchesToExclude;
 
-        static CecilSymbolHelper()
+        public CecilSymbolHelper()
         {
             // Create single instance, we cannot collide because we use full method name as key
-            CompilerGeneratedBranchesToExclude = new ConcurrentDictionary<string, int[]>();
+            _compilerGeneratedBranchesToExclude = new ConcurrentDictionary<string, int[]>();
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Symbols/CecilSymbolHelper.cs:27]: We can initialize inline and remove ctor, before we wanted to be sure single thread initialization, now we use an singleton instance so no double init issue. Keep the comment above pls.

File: src/coverlet.console/Program.cs [hunk 8]
```diff
@@ -88,7 +90,8 @@ static int Main(string[] args)
                     logger,
                     serviceProvider.GetService<IInstrumentationHelper>(),
                     fileSystem,
-                    sourceTranslator);
+                    sourceTranslator,
```
Comment (MarcoRossignoli) [inline src/coverlet.console/Program.cs:93]: can we use `serviceProvider.GetRequiredService` everywhere where we don't use variable for more than one place?I see some assignation above, `sourceTranslator` is used in more than one place?

File: src/coverlet.core/Symbols/CecilSymbolHelper.cs (hunk 7 above)
Comment (daveMueller) [reply src/coverlet.core/Symbols/CecilSymbolHelper.cs]: Done

File: src/coverlet.console/Program.cs (hunk 8 above)
Comment (daveMueller) [reply src/coverlet.console/Program.cs]: Done

**Pull request merged**